lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -26,37 +26,49 @@
 	 ************************************/
 
 	/**
-	 * Get the presence state for this Channel. Implicitly attaches the
-	 * Channel. However, if the channel is in or moves to the FAILED
-	 * state before the operation succeeds, it will result in an error
-	 * @return: the current present members.
+	 * String parameter names to get() call with Param[] as an argument
+	 */
+	public final static String GET_WAITFORSYNC = "waitForSync";
+	public final static String GET_CLIENTID = "clientId";
+	public final static String GET_CONNECTIONID = "connectionId";
+
+	/**
+	 * Get the presence state for this channel. Take Param[] array as an argument
+	 * @param params
+	 * @return
 	 * @throws AblyException
+	 * @throws InterruptedException
 	 */
-	public synchronized PresenceMessage[] get() throws AblyException {
+	public synchronized PresenceMessage[] get(Param... params) throws InterruptedException, AblyException {
```

Comment:
According to spec `Presence.get()` without argument is a blocking call as well and can throw `InterruptedException` as well. So yes, compatibility will be broken in a lot of places. On the other hand `InterruptedException` is a well established standard in Java for blocking calls